core/src/main/java/org/ehcache/core/spi/cache/AbstractValueHolder.java

```diff
@@ -109,7 +124,16 @@ public long lastAccessTime(TimeUnit unit) {
   }
 
   public void setLastAccessTime(long lastAccessTime, TimeUnit unit) {
-    this.lastAccessTime = unit.convert(lastAccessTime, nativeTimeUnit());
+    lastAccessTime = unit.convert(lastAccessTime, nativeTimeUnit());
+    while (true) {
+      long read = this.lastAccessTime;
```

Comment:
Renaming the `read` variable `current` and the local `lastAccessTime` one `update` would make this method trivial to mentally match it with the `updateExpirationTime()` one. That would help future maintenance.